finesse/gui/stepper_motor_view.py

```diff
@@ -0,0 +1,59 @@
+"""Code for controlling the stepper motor which moves the mirror."""
+from pubsub import pub
+from PySide6.QtWidgets import (
+    QButtonGroup,
+    QGridLayout,
+    QGroupBox,
+    QPushButton,
+    QSpinBox,
+)
+
+from ..config import ANGLE_PRESETS
+
+
+class StepperMotorControl(QGroupBox):
+    """A control showing buttons for moving the mirror to a target."""
+
+    def __init__(self) -> None:
+        """Create a new StepperMotorControl."""
+        super().__init__("Target control")
+
+        layout = QGridLayout()
+
+        # Bundle all the buttons for moving the mirror into one group
+        self.button_group = QButtonGroup()
+        self.button_group.buttonClicked.connect(self._preset_clicked)  # type: ignore
+
+        # Add all the buttons for preset positions
+        BUTTONS_PER_ROW = 4
+        for i, preset in enumerate(ANGLE_PRESETS):
+            btn = self._add_checkable_button(preset.upper())
+            self.button_group.addButton(btn)
+
+            row, col = divmod(i, BUTTONS_PER_ROW)
+            layout.addWidget(btn, row, col)
+
+        # We also have a way for users to move the mirror to an angle of their choice
+        self.angle = QSpinBox()
+        self.angle.setMaximum(359)
+        self.goto = self._add_checkable_button("GOTO")
+
+        layout.addWidget(self.angle, 1, 2)
+        layout.addWidget(self.goto, 1, 3)
+
+        self.setLayout(layout)
+
+    def _add_checkable_button(self, name: str) -> QPushButton:
+        """Add a selectable button to button_group."""
+        btn = QPushButton(name)
+        btn.setCheckable(True)
+        btn.setStyleSheet("QPushButton::checked{background: blue; color: white;}")
```

Comment:
Unless you fancy specifically this style, buttons in a group change colour when they are pushed - at least in Mac; I would expect the same in others. See video. So I think this line can be removed.


https://user-images.githubusercontent.com/6095790/205260718-a79fd9c1-008e-48f1-a0b0-7e7ace8a1f28.mov

Reply:
Good point. I went with blue because that was the colour used in the old GUI, but I think the default style is probably fine.

For me, the buttons just appear as pressed in by default, not blue, but I guess that's fine too.